clusterloader2/pkg/framework/client/objects.go

```diff
@@ -172,7 +172,9 @@ func PatchObject(dynamicClient dynamic.Interface, namespace string, name string,
 func DeleteObject(dynamicClient dynamic.Interface, gvk schema.GroupVersionKind, namespace string, name string) error {
 	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
 	deleteFunc := func() error {
-		return dynamicClient.Resource(gvr).Namespace(namespace).Delete(name, nil)
+		falseVar := false
+		deleteOption := &metav1.DeleteOptions{OrphanDependents: &falseVar}
```

Comment:
This means that deletion is no longer a default k8s deletion.

I'm fine with that semantic, but we need to be very clear about it:
- in comments in this file
- in design doc (it should be updated together with this PR)
- additionally in readme too

Reply:
Added in 3rd commit.